### PR TITLE
Use CustomComparator for a object in list 492

### PR DESCRIPTION
**Pull request by pszymczyk**
It does not work wit Sets at the moment

### CODE REVIEW

File: javers-core/src/test/groovy/org/javers/core/CustomPropertyComparatorE2ETest.groovy [hunk 1]
```diff
@@ -87,4 +88,26 @@ class CustomPropertyComparatorE2ETest extends Specification {
...
+        then:
+        javers.compare(left, right).changes.size() == 1
+
+        when:
+        javers = JaversBuilder.javers()
+                .registerValueChangeCustomComparator(new DummyCustomPropertyComparator(), String)
+                .build()
+
+        then:
+        javers.compare(left, right).changes.size() == 0
```
Comment (bartoszwalacik) [inline javers-core/src/test/groovy/org/javers/core/CustomPropertyComparatorE2ETest.groovy]: So we only test that the new feature **doesn't** work?

File: javers-core/src/main/java/org/javers/core/JaversBuilder.java [hunk 2]
```diff
@@ -477,6 +479,37 @@ public JaversBuilder withObjectAccessHook(ObjectAccessHook objectAccessHook) {
...
+     * Internally, given type is mapped as {@link CustomType}.
+     * <br/><br/>
+     *
+     * Custom types are serialized to JSON using Gson defaults.
+     *
+     * @param <T> custom type
+     * @param customType class literal to define a custom type
+     * @see CustomPropertyComparator
+     */
+    public <T> JaversBuilder registerValueChangeCustomComparator(CustomPropertyComparator<T, ValueChange> comparator, Class<T> customType){
```
Comment (bartoszwalacik) [inline javers-core/src/main/java/org/javers/core/JaversBuilder.java]: The new API is unclear for me. Why Custom Comparator for Collections of Values returns `ValueChange`?
When we compare Collections we produce  `ContainerChange` with list of entry changes (add, remove, etc).

Second, does it affect *normal* comparing of Values when they are simple properties and not elements of a Collection?

I think, that we should somehow extend the `CustomPropertyComparator` interface to support Collections,
and let user to register one CustomPropertyComparator per one Custom type

File: javers-core/src/main/java/org/javers/core/diff/custom/CustomValueComparator.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import org.javers.core.JaversBuilder;
+import org.javers.core.metamodel.type.ValueType;
+
+/**
+ * Can be used only for comparing {@link ValueType}.
+ *
+ * @see JaversBuilder#registerValue(Class, CustomValueComparator)
+ */
+@FunctionalInterface
+public interface CustomValueComparator<T> {
```
Comment (pszymczyk) [inline javers-core/src/main/java/org/javers/core/diff/custom/CustomValueComparator.java]: very nice 👍

**Comment by pszymczyk** [comment]
+1

**Pull request merged**